### PR TITLE
Moved fuse array functions from MPQC to TA

### PR DESCRIPTION
**Pull request by kmp5VT**
Moved functions that can transform TA array to vectors of TA arrays and back from MPQC to TA.

### CODE REVIEW

File: src/TiledArray/conversions/fuse_arrays.h [hunk 1]
```diff
@@ -0,0 +1,220 @@
+//
+// Created by Chong Peng on 2019-05-01.
+//
+
+#ifndef SRC_MPQC_MATH_EXTERNAL_TILEDARRAY_FUSE_ARRAYS_H_
```
Comment (pchong90) [inline src/TiledArray/conversions/fuse_arrays.h]: header guard is still in MPQC format, need to adjust to TiledArray format, check how other TiledArray headers define header guards

File: src/TiledArray/conversions/fuse_arrays.h [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+#define SRC_MPQC_MATH_EXTERNAL_TILEDARRAY_FUSE_ARRAYS_H_
+
+#include "mpqc/math/external/tiledarray/tiledarray.h"
```
Comment (pchong90) [inline src/TiledArray/conversions/fuse_arrays.h]: this is path is still for mpqc library, need to update

File: src/TiledArray/conversions/fuse_arrays.h [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+namespace mpqc {
```
Comment (pchong90) [inline src/TiledArray/conversions/fuse_arrays.h]: change to TiledArray namespace

File: src/TiledArray/conversions/fuse_arrays.h [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+          make_tile, split_array.trange().make_tile_range(index),
+          fused_array.find(fused_array_index));
+      split_array.set(index, new_tile);
+    }
+  }
+
+  return split_array;
+}
+
+}  // namespace mpqc
```
Comment (pchong90) [inline src/TiledArray/conversions/fuse_arrays.h]: update namespace

File: src/TiledArray/conversions/fuse_arrays.h [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+#endif  // SRC_MPQC_MATH_EXTERNAL_TILEDARRAY_FUSE_ARRAYS_H_
```
Comment (pchong90) [inline src/TiledArray/conversions/fuse_arrays.h]: update header guard

**Comment by evaleev** [comment]
Good points from Chong. But... I am renaming things locally ... Hold off any changes.

**Comment by kmp5VT** [comment]
It looks like I made these changes in an install copy and forgot to put them in the source.  Thanks for looking over the files Chong

**Comment by evaleev** [comment]
I renamed a pretty much everything ... @kmp5VT satisfy Chong's requests + add a simple unit test (only for functions outside `detail`).

**Pull request merged**